releng: Consolidation of Patch Release Team and Branch Managers

justaugustus opened the pull request:
#### What type of PR is this:

/kind cleanup documentation

#### Action items

- [ ] (https://github.com/kubernetes/org/pull/1919) Consolidate GitHub teams
- [ ] Update OWNERS
  - [ ] (https://github.com/kubernetes/sig-release/pull/1106) k/sig-release
  - [ ] (https://github.com/kubernetes/release/pull/1336) k/release
- [ ] (https://github.com/kubernetes/test-infra/pull/17868) Update bot comment for patch releases
- [ ] (https://github.com/kubernetes/website/pull/21548) Update references to the patch release process on k/website
- [ ] (https://github.com/kubernetes/security/pull/90) Update Release Manager roles on k/security

#### What this PR does / why we need it:

From a [note I sent](https://groups.google.com/a/kubernetes.io/d/topic/release-managers/_qqdJre-JQY/discussion) to @kubernetes/release-engineering earlier in the week:

> Hey Release Managers,
> 
> I had a chat with Tim yesterday about something that's been on my mind a bit.
> 
> From an access perspective, the Patch Release Team and Branch Managers are nearly identical:
> 
>     * Authorization/access to stage/release Kubernetes
> 
>     * Write access to kubernetes/kubernetes
> 
>     * In the notification circle for active CVEs and responsible for helping to mitigate CVEs
> 
> 
> I think the Branch Managers (Carlos, Sascha, and Dan) have been stepping up in a big way to provide coverage for patch releases, and I see no real reason to not give them the mandate to cut patch releases on a consistent/official basis.
> 
> That said, I'd like to consolidate the Patch Release Team + Branch Managers into simply "Release Managers", with the Release Manager Associates remaining as the apprentice-level members of the team.
> 
> There are a few things to think through (and we can step through that at a later time), but I figured it'd be good to start the conversation.

/hold for discussion
/assign @tpepper @saschagrunert @cpanato @hasheddan 

#### Which issue(s) this PR fixes:

<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->

Fixes #

#### Special notes for your reviewer:

saschagrunert commented:
To me it makes totally sense to unify both roles. 👍 We also want to unify the release tooling and diverging code paths between minor and patch versions already caused trouble in the past.

idealhack commented:
+1, this makes sense to me

markyjackson-taulia commented:
+1 extremely logical

cpanato commented:
+1

hasheddan commented:
+1

justaugustus commented:
First PR up: https://github.com/kubernetes/org/pull/1919

justaugustus commented:
Fixed up the Release Manager onboarding template and added a content notice for handbooks, so I think this is good to go now.

I've also updated the action items in the PR description with the following PRs:

- Update OWNERS
  - (https://github.com/kubernetes/sig-release/pull/1106) k/sig-release
  - (https://github.com/kubernetes/release/pull/1336) k/release
- (https://github.com/kubernetes/test-infra/pull/17868) Update bot comment for patch releases
- (https://github.com/kubernetes/website/pull/21548) Update references to the patch release process on k/website
- (https://github.com/kubernetes/security/pull/90) Update Release Manager roles on k/security

tpepper commented on .github/ISSUE_TEMPLATE/release-manager.md:
```diff
@@ -9,9 +9,9 @@ labels: sig/release, area/release-eng
 e.g., (at)example_user
 
-### [Release Manager](/release-managers.md) role
+### [Release Manager](https://git.k8s.io/sig-release/release-managers.md) role
```
Would it feel too grammatically strange to have this be "Release Managers" throughout for consistency?

tpepper commented:
> I think we should do the de-dup and handbook cleanup in this PR. I think it's almost entirely deletions. That would be better than adding the three warnings about out of date content to be fixed up later.

To be more explicit, I support a "De-dup" in this PR which is just deleting the patch-release-team handbook.  The operation of making a release is covered by other docs.  The obscure bits like the spreadsheets need re-evaluated as they're not fully functioning on team basis today.

justaugustus commented:
Thanks for the reviews everyone!
De-dupe to fast-follow.
/hold cancel
